LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-forEach.any.js

```diff
@@ -86,82 +86,106 @@ promise_test(async (t) => {
+                     "rejected");
+  }, () => {
+    // See the documentation above. The rejection handler (i.e., this code) for
+    // immediately-aborted operator Promises runs before any later-scheduled
+    // Promise resolution/rejections.
+    assert_false(postSubscriptionPromiseResolved,
+        "Operator promise rejects before later promise");
+  });
+  const postSubscriptionPromise =
+      Promise.resolve().then(() => postSubscriptionPromiseResolved = true);
 
-  // After a single task, assert that everything has happened correctly (and
-  // incrementally in the right order);
-  await new Promise(resolve => {
-    t.step_timeout(resolve);
+  await rejectedPromise;
+}, "forEach(): Subscribing with an aborted signal returns an immediately " +
+   "rejected promise");
+
+// See https://github.com/WICG/observable/issues/96 for discussion about this.
+promise_test(async () => {
+  const results = [];
+
+  const observable = new Observable(subscriber => {
+    results.push(`Subscribed. active: ${subscriber.active}`);
+
+    subscriber.signal.addEventListener('abort', e => {
+      results.push("Inner signal abort event");
+      Promise.resolve("Inner signal Promise").then(value => results.push(value));
+    });
+
+    subscriber.addTeardown(() => {
+      results.push("Teardown");
+      Promise.resolve("Teardown Promise").then(value => results.push(value));
+    });
   });
-  assert_true(outerAbortEventMicrotaskRun,
-      "Final: outer abort microtask has fired");
-  assert_true(forEachPromiseRejectionMicrotaskRun,
-      "Final: forEach rejection microtask has fired");
-  assert_true(innerAbortEventMicrotaskRun,
-      "Final: inner abort microtask has fired");
-  assert_equals(rejectionError, error, "Promise is rejected with the right " +
-      "value");
-}, "forEach visitor callback rejection microtask ordering");
+
+  const controller = new AbortController();
+  controller.signal.addEventListener('abort', e => {
+    results.push("Outer signal abort event");
+    Promise.resolve("Outer signal Promise").then(value => results.push(value));
+  });
+
+  // Subscribe.
+  observable.forEach(() => { }, {signal: controller.signal});
+  controller.abort();
+
+  assert_array_equals(results, [
+    "Subscribed. active: true",
+    "Inner signal abort event",
+    "Teardown",
+    "Outer signal abort event",
+  ], "Events and teardowns are fired in the right ordered");
+
+  // Everything microtask above should be queued up by now, so queue one more
+  // final microtask that will run after all of the others, wait for it, and the
+  // check `results` is right.
+  await Promise.resolve();
+  assert_array_equals(results, [
+    "Subscribed. active: true",
+    "Inner signal abort event",
+    "Teardown",
+    "Outer signal abort event",
+    "Inner signal Promise",
+    "Teardown Promise",
+    "Outer signal Promise",
+  ], "Promises resolve in the right order");
+}, "forEach(): Operator Promise abort ordering");
```

Comment:
Note for reviewer: This is a carbon copy of the abort ordering test for `toArray`.